scico/linop/optics.py

```diff
@@ -59,13 +59,15 @@ def radial_transverse_frequency(
         raise ValueError("Invalid input dimensions; must be 1 or 2")
 
     if np.isscalar(dx):
-        dx = (dx,) * ndim
+        dx = (dx,) * ndim  # type: ignore
```

Comment:
Error suppression and `assert` below needed because `mypy` does not appear to be capable of understanding the implications of the conditional here.

Reply:
Possibly this could be achieved via a type guarding version of `isscalar` https://mypy.readthedocs.io/en/latest/type_narrowing.html#user-defined-type-guards